src/gleam/time/timestamp.gleam

```diff
@@ -227,12 +265,388 @@ fn to_civil(minutes: Int) -> #(Int, Int, Int) {
   #(year, month, day)
 }
 
-// TODO: docs
-// TODO: test
-// TODO: implement
-// pub fn parse_rfc3339(input: String) -> Result(Timestamp, Nil) {
-//   todo
-// }
+/// Parses an RFC 3339 formatted time string into a `Timestamp`.
+/// 
+/// # Examples
+///
+/// ```gleam
+/// let assert Ok(ts) = parse_rfc3339("1970-01-01T00:00:01.12345678999Z")
+/// to_unix_seconds_and_nanoseconds(ts)
+/// // -> #(1, 123_456_789)
+/// 
+/// let assert Ok(ts) = parse_rfc3339("2025-01-10t15:54:30-05:15")
+/// to_unix_seconds_and_nanoseconds(ts)
+/// // -> #(1_736_543_370, 0)
+/// 
+/// let assert Error(Nil) = timestamp.parse_rfc3339("1995-10-31")
+/// ```
+///
+/// # Notes
+/// 
+/// - Follows the grammar specified in section 5.6 Internet Date/Time Format of 
+///   RFC 3339 (https://datatracker.ietf.org/doc/html/rfc3339#section-5.6).
+/// - The `T` and `Z` characters may alternatively be lower case `t` or `z`, 
+///   respectively.
+/// - Full dates and full times must be separated by `T` or `t`, not any other 
+///   character such as a space (` `).
+/// - Leap seconds rules are not considered.  That is, any timestamp may 
+///   specify digts `00` - `60` for the seconds.
+/// - Any part of a fractional second that cannot be represented in the 
+///   nanosecond precision is tructated.  That is, for the time string, 
+///   `"1970-01-01T00:00:00.1234567899Z"`, the fractional second `.1234567899` 
+///   will be represented as `123_456_789` in the `Timestamp`.
+/// 
```

Comment:
Do we need to describe the format here? Seems some of this is redundant?

Reply:
Yes I see what you mean.  These notes are trying to be specific about things that are not really specified in the RFC.  E.g., it says applications may choose to use a space or something else to separate date and time, or that case of T and Z may be considered, and the bit about truncating.

Maybe it would be better in the readme rather than here? Or, if you think it's better, it could be left "unspecified".